dashboard/app/models/plc/user_course_enrollment.rb

```diff
@@ -25,6 +26,8 @@ class Plc::UserCourseEnrollment < ActiveRecord::Base
   validates :user, presence: true
   validates :plc_course, presence: true
 
+  validates_uniqueness_of :user_id, scope: :plc_course_id
```

Comment:
you can save a validation step (assuming you never change it after creating it) by doing this only `on: create`